jailer/src/lib.rs

```diff
@@ -332,19 +332,16 @@ fn sanitize_process() {
     }
 }
 
-fn open_dev_kvm() -> Result<i32> {
-    // Safe because we use a constant null-terminated string and verify the result.
-    let ret = unsafe { libc::open("/dev/kvm\0".as_ptr() as *const libc::c_char, libc::O_RDWR) };
+fn open_dev_kvm() -> Result<()> {
+    // The following unwrap will only fail when `/dev/kvm` cannot be open in which case
+    // it is better to panic and finish the execution.
+    let kvm_fd = Kvm::open(false).unwrap();
 
-    if ret < 0 {
-        return Err(Error::OpenDevKvm(sys_util::Error::last()));
+    if kvm_fd != KVM_FD {
```

Comment:
would it make sense to treat this case inside the `Kvm::open()` method ?

Reply:
From my understanding FDs are assigned in incremental order for each process. So the KVM_FD number being 3 is specific to the Firecracker process. The fd numbers 0, 1, 2 are reserved for STDIN, STDOUT, STDERR. When we open /dev/kvm it will have the fd number 3 as it is the next one that's available. Now, because kvm is going to be an independent crate, moved to rust-vmm we cannot make the assumption that other project will open the kvm fd first before opening any other fds so we cannot make the assumption that kvm fd will always be the 3rd one.